src/Intersection.jl

```diff
@@ -475,8 +490,38 @@ end # _line_search
 end # quote
 end # load_optim
 
-# Symmetric case
-ρ(ℓ::AbstractVector{N}, cap::Intersection{N, S, <:LazySet};
-  algorithm::String="line_search", check_intersection::Bool=true,
-  kwargs...) where {N<:AbstractFloat, S<:Union{HalfSpace, Hyperplane}} = ρ(ℓ, cap.Y ∩ cap.X;
-  algorithm=algorithm, check_intersection=check_intersection, kwargs...)
+"""
+    _projection(ℓ, X, H::Union{Hyperplane{N}, Line{N}}; kwargs...) where {N}
+
+Given a compact and convex set ``X`` and a hyperplane ``H = \\{x: n ⋅ x = γ \\}``,
+calculate ...
+
+### Input
+
+- `ℓ`      -- direction
+- `X`      -- set
+- `H`      -- hyperplane
+
+### Output
+
+The support function of ``X ∩ H`` along direction ``ℓ``.
+"""
+function _projection(ℓ, X, H::Union{Hyperplane{N}, Line{N}};
+                     lazy_linear_map=false,
+                     lazy_2d_intersection=true,
+                     kwargs...) where {N}
+
+    options = Dict(kwargs)
+    n = H.a                  # normal vector to the hyperplane
+    γ = H.b                  # displacement of the hyperplane
+    Πnℓ = vcat(n', ℓ')       # projection map
+
+    x_dir = [one(N), zero(N)]
+    y_dir = [zero(N), one(N)]
+
+    Lγ = Line(x_dir, γ)
+
+    Snℓ = lazy_linear_map ? LinearMap(Πnℓ, X) : linear_map(Πnℓ, X)
```

Comment:
In the spirit of above, what about `\bigcap<TAB>Sℓ` or `S\bigcap<TAB>ℓ` (resp. `d` instead of `ℓ`)?

Reply:
nice! we can use the `\bigcap` notation in the hybrid algorithm too 👍

Reply:
hmm better if i use `Xnℓ` since the given set is `X`

Reply:
What does the notation mean?